File: logging/logging.go
package logging

import (
	"io"
	"log"
	"os"
	"strings"
)

const (
	levelInfo  = "info"
	levelError = "error"
)

var currentLevel = levelInfo

var (
	infoLogger  = log.New(os.Stdout, "", log.LstdFlags)
	errorLogger = log.New(os.Stderr, "", log.LstdFlags)
)

func SetLevel(level string) {
	level = strings.ToLower(level)
	switch level {
	case levelInfo, levelError:
		currentLevel = level
	default:
		currentLevel = levelInfo
	}
	infoLogger.Printf("log level set to %s", currentLevel)
}

func SetOutputs(infoOut, errOut io.Writer) {
	if infoOut != nil {
		infoLogger.SetOutput(infoOut)
	}
	if errOut != nil {
		errorLogger.SetOutput(errOut)
	}
}

func Info(msg string) {
	if currentLevel == levelInfo {
		infoLogger.Println("[INFO] " + msg)
	}
}

func Infof(format string, args ...interface{}) {
	if currentLevel == levelInfo {
		infoLogger.Printf("[INFO] "+format, args...)
	}
}

func Error(msg string) {
	errorLogger.Println("[ERROR] " + msg)
}

func Errorf(format string, args ...interface{}) {
	errorLogger.Printf("[ERROR] "+format, args...)
}

File: src/logging/logging.ts
export const LevelInfo = "info";
export const LevelError = "error";

let currentLevel = LevelInfo;

export function setLevel(level: string): void {
  level = level.toLowerCase();
  if (level !== LevelInfo && level !== LevelError) {
    level = LevelInfo;
  }
  currentLevel = level;
  console.log(`Log level set to ${currentLevel}`);
}

export function info(msg: string): void {
  if (currentLevel === LevelInfo) {
    console.log(`[INFO] ${msg}`);
  }
}

export function infof(format: string, ...args: any[]): void {
  if (currentLevel === LevelInfo) {
    console.log(`[INFO] ${format}`, ...args);
  }
}

export function error(msg: string): void {
  console.log(`[ERROR] ${msg}`);
}

export function errorf(format: string, ...args: any[]): void {
  console.log(`[ERROR] ${format}`, ...args);
}
